tests/phpunit/test-class-wc-dintero-hp-checkout.php

```diff
@@ -59,20 +68,67 @@ public function test_process_cart_with_discount()
 		update_post_meta( $new_coupon_id, 'free_shipping', 'no' );
 
 		WC()->initialize_cart();
-		$product = WC_Helper_Product::create_simple_product(true, array(
-			'price' => 179.00,
-			'regular_price' => 179.00,
-		));
 
+		foreach ( $products as $item ) {
+			$product = WC_Helper_Product::create_simple_product(true, array(
+				'price' => $item['price'],
+				'regular_price' => $item['regular_price'],
+			));
+			$cart_item_id = WC()->cart->add_to_cart($product->get_id(), $item['qty']);
+			$cart_item = WC()->cart->get_cart_item($cart_item_id);
+			$cart_item['line_tax'] = $tax_amount;
+		}
 
-		$cart_item_id = WC()->cart->add_to_cart($product->get_id(), 4);
-
-		$cart_item = WC()->cart->get_cart_item($cart_item_id);
 		WC()->cart->add_discount( sanitize_text_field( $coupon_code ));
-		$cart_item['line_tax'] = 114.56;
-
 		$checkout = new WC_Dintero_HP_Checkout();
 		$checkout->process_cart();
-		$this->assertEquals(57280, reset($checkout->order_lines)['amount']);
+		foreach ($checkout->order_lines as $item) {
+			$this->assertEquals($result, $item['amount']);
+		}
+	}
+
+	/**
+	 * @return array[]
+	 */
+	public function process_cart_with_discount_dataprovider()
+	{
+		return array(
+			array(
+				'products' => array(
+					array(
+						'sku' => 'dummy1',
+						'price' => 179.00,
+						'regular_price' => 179.00,
+						'qty' => 4,
+					)
+				),
+				'coupon_code' => 'dintero20',
+				'discount_amount' => '20',
+				'discount_type' => 'percent',
+				'tax_amount' => 114.56,
+				'result' => 57280
+			),
+			array(
```

Comment:
The test runs without problems even without the changes in `class-wc-dintero-hp-checkout.php`. Did you test it manually?